src/main/scala/com/datamountaineer/streamreactor/connect/schemas/SinkRecordConverterHelper.scala

```diff
@@ -0,0 +1,186 @@
+                payloadSchema = record.valueSchema(),
+                key = false,
+                fields = fields,
+                ignoreFields = ignoreFields)
+      } else {
+        logger.warn(
+          s"Value is null for topic [${record.topic()}], partition [${record
+            .kafkaPartition()}], offset [${record.kafkaOffset()}])")
+        new Struct(SchemaBuilder.struct().build())
+      }
+
+      //create a new struct with the keys, values and headers
+      val struct = keyStruct ++ valueStruct ++ structFromHeaders(headerFields)
+
+      new SinkRecord(
+        record.topic(),
+        record.kafkaPartition(),
+        if (retainKey) record.keySchema() else null,
+        if (retainKey) record.key() else null,
+        struct.schema(),
+        struct,
+        record.kafkaOffset(),
+        record.timestamp(),
+        record.timestampType(),
+        if (retainHeaders) record.headers() else new ConnectHeaders()
+      )
+    }
+
+    private def extract(payload: Object,
+                        payloadSchema: Schema,
+                        key: Boolean,
+                        fields: Map[String, String],
+                        ignoreFields: Set[String]): Struct = {
+
+      if (payloadSchema == null) {
+        //json with no schema
+        val j: util.Map[String, Any] = convertSchemalessJson(
+          record = record,
+          fields = fields,
+          ignoreFields = ignoreFields,
+          key = key,
+          includeAllFields = fields.contains("*")
+        )
+
+        val newSchema = SchemaBuilder.struct()
+        j.asScala
+          .foreach {
+            case (name, value) => newSchema.field(name, Schema.STRING_SCHEMA)
+          }
+
+        val newStruct = new Struct(newSchema.build())
+        j.asScala.foreach {
+          case (name, value) => newStruct.put(name, value.toString)
+        }
+        newStruct
+
+      } else {
+        payloadSchema.`type`() match {
+          //struct
+          case Schema.Type.STRUCT =>
+            payload
+              .asInstanceOf[Struct]
+              .reduceToSchema(schema = payloadSchema,
+                              fields = fields,
+                              ignoreFields = ignoreFields)
+
+          // json with string schema
+          case Schema.Type.STRING =>
+            val j = convertStringSchemaAndJson(record = record,
+                                               fields = fields,
+                                               ignoreFields = ignoreFields,
+                                               key = key,
+                                               includeAllFields =
+                                                 fields.contains("*"))
+
+            val newSchema = SchemaBuilder.struct()
+            val jFields = j.asInstanceOf[JObject].values
+            jFields.foreach {
+              case (name, value) =>
+                // default to string
+                newSchema.field(name, Schema.STRING_SCHEMA)

```

Comment:
Same as above. Why default to STRING? what if one has `amount:123.4` which is Double or BigDecimal (depending of the Json deserialization settings)